### ISSUE TITLE
NSTaggedPointerString error

### ISSUE THREAD

**Issue by marcelkipp** [BranchMetrics/cordova-ionic-phonegap-branch-deep-linking-attribution#671]
I have an issue with using the sendCommerceEvent-Method of the module. As soon as I'm calling the method my ios app is crashing with the following error: 

I am calling the method as following in my component:

**NSTaggedPointerString countByEnumeratingWithState** in Line 320 in the file BranchSDK.m

`var metadata = {
      customerEventAlias: 'Purchase Event',
      transactionID: this.buyInfo.transaction.original_transaction_id,
      currency: this.buyInfo.currency,
      revenue: this.buyInfo.prices,
      shipping: 0,
      tax: 0,
      coupon: 'test_coupon',
      affiliation: 'test_affiliation',
      description: 'Test add to cart event',
      searchQuery: 'test keyword',
      customData: this.buyInfo
    };

    this.branch.sendCommerceEvent("PURCHASE", metadata);`

Does anyone had this issue before or do I send wrong data to the plugin?

**Comment by felipeclopes** [comment]
We're experience the same thing and are not able to track events like TRIAL_STARTED and PURCHASE.

Are there any updates in this issue?

**Comment by marcelkipp** [comment]
@felipeclopes No, unfortunately no updated. We decided to remove branch from our project.

**Comment by adrianyg7** [comment]
Hey @marcelkipp @felipeclopes Can you provide additional information about your build environment?
Cordova version, branch-cordova-sdk version, node version, Xcode version, installed plugins.

Please note that `sendCommerceEvent` method is deprecated since v.3.1.0
Refer to https://help.branch.io/developers-hub/docs/cordova-phonegap-ionic#track-commerce for tracking commerce events.

**Comment by felipeclopes** [comment]
I created this [PR](https://github.com/ionic-team/ionic-native/pull/3675) fixing the issue on [IonicNative](https://github.com/ionic-team/ionic-native) - so now the IonicNative Branch library has the correct method `sendBranchEvent` implementation.

**Comment by felipeclopes** [comment]
@echo-branch this issue can be closed.
